### PR TITLE
Configure Scala runtime configurations as runtime classpaths

### PR DESCRIPTION
**Pull request by rieske**
This is preparation work before deprecating resolution of
configurations that do not have attributes - before we
can do that, we should ensure that our own resolvable
configurations have attributes configured.

https://github.com/gradle/gradle/issues/13736

### CODE REVIEW

File: subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java [hunk 1]
```diff
@@ -122,7 +122,10 @@ private void configureConfigurations(final Project project, final Usage incremen
 
         Configuration zinc = project.getConfigurations().create(ZINC_CONFIGURATION_NAME);
         zinc.setVisible(false);
+        zinc.setCanBeConsumed(false);
```
Comment (rieske) [inline subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java:125]: This is technically a breaking change. We could instead first deprecate this for consumption and delay attributes configuration for 8.0. Thoughts?

Comment (jjohannes) [reply subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java:125]: I would think this "breakage" is ok, as in practice this should not be consumed by anyone. And if, it's easy to work around. (e.g. by just setting it back to `true` in your build script.)
We can add a sentence to the "potentially breaking changes" section of the upgrading guide.

File: subprojects/scala/src/main/java/org/gradle/api/tasks/ScalaRuntime.java [hunk 2]
```diff
@@ -59,9 +62,11 @@
     private static final Pattern SCALA_JAR_PATTERN = Pattern.compile("scala-(\\w.*?)-(\\d.*).jar");
 
     private final Project project;
+    private final JvmEcosystemUtilities jvmEcosystemUtilities;
 
     public ScalaRuntime(Project project) {
         this.project = project;
+        this.jvmEcosystemUtilities = ((ProjectInternal) project).getServices().get(JvmEcosystemUtilities.class);
```
Comment (rieske) [inline subprojects/scala/src/main/java/org/gradle/api/tasks/ScalaRuntime.java:69]: Injecting this service via constructor is possible, but this class is in the public API and changing this constructor breaks binary compatibility.

File: subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java (hunk 1 above)
Comment (rieske) [reply subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java]: I reverted the changes to ScalaBasePlugin, leaving only non-breaking ones in this PR.
ScalaBasePlugin's zinc configuration I will handle in a separate change where I will deprecate it for consumption first and will then add required attributes together with a capability-workaround.

**Review by ljacomet — APPROVED** [review]

**Pull request merged**